packages/payment-detection/src/any/any-to-erc20-proxy.ts

```diff
@@ -134,23 +129,24 @@ export class AnyToERC20PaymentDetector extends ERC20FeeProxyPaymentDetectorBase<
           conversionProxyContractAddress,
           conversionProxyCreationBlockNumber,
           conversionProxyAbi,
-          toAddress,
+          address,
           eventName,
-          network,
+          paymentChain,
           acceptedTokens,
           maxRateTimespan,
         );
 
-    const events = await infoRetriever.getTransferEvents();
-
-    const balance = events
-      .reduce((acc, event) => acc.add(BigNumber.from(event.amount)), BigNumber.from(0))
-      .toString();
+    return infoRetriever.getTransferEvents() as Promise<
+      PaymentTypes.IPaymentNetworkEvent<PaymentTypes.IERC20FeePaymentEventParameters>[]
+    >;
+  }
 
-    return {
-      balance,
-      events,
-    };
+  protected getPaymentChain(request: RequestLogicTypes.IRequest): string {
+    const network = this.getPaymentExtension(request).values.network;
+    if (!network) {
+      throw Error(`request.extensions[${this.paymentNetworkId}].values.network must be defined`);
+    }
+    return network;
```

Comment:
Looks duplicated with the one in `any-to-eth-proxy.ts`. Is it possible to move it to `FeeReferenceBasedDetector`?

Reply:
it's duplicated, yes, but cannot be moved to FeeReferenceBasedDetector, because it's not the default behavior for this, it's specific to `any-to-xxx`, but these classes don't have a common base (maybe they should... but AnyToERC20 already inherits `ERC20FeeProxyPaymentDetectorBase` which also makes sense...)